fvt/storage/storage_test.go

```diff
@@ -0,0 +1,189 @@
+			// The controller will update the deployment with the pvc_mount, but
+			// if the old runtime pods are still around, the ISVC will get deployed
+			// on an old runtime pod without the PVC mounted and keep failing
+			// until a new pod with the PVC is ready and the controller finally
+			// decides to move the ISVC onto the new pod that has the PVC mounted.
+			// However, this process can take a long time (how long?) so, we take
+			// some extra measures to increase our chances for quick success:
+			// - scale to 0 prohibits the new ISVC to land on an old runtime pod
+			//   that does not have the "any" PVC mounted yet
+			// - use more than 1 pod per runtime so controller will not kill new
+			//   pods that have the PVC mounted but because the ISVC is loaded on
+			//   the old pod (without the PVC) but the old pod gets kept around
+			//   instead of the new because the ISVC is still on there -- even
+			//   though its failing
+			// - allowAnyPVC needs rest-proxy enabled (not sure why)
+			config := map[string]interface{}{
+				"allowAnyPVC":    true,
+				"podsPerRuntime": 1,
+				"scaleToZero": map[string]interface{}{
+					"enabled": true,
+				},
+				"restProxy": map[string]interface{}{
+					"enabled": true,
+				},
+			}
+			By("Updating the user config to allow any PVC")
+			FVTClientInstance.ApplyUserConfigMap(config)
+
+			// TODO: separate test case for not having storage secret at all, currently not working
+			// Events:
+			//  Type     Reason       Age                  From               Message
+			//  ----     ------       ----                 ----               -------
+			//  Normal   Scheduled    3m23s                default-scheduler  Successfully assigned modelmesh-serving/modelmesh-serving-mlserver-0.x-54685b95d5-6xmck to 10.87.76.74
+			//  Warning  FailedMount  80s                  kubelet            Unable to attach or mount volumes: unmounted volumes=[storage-config], unattached volumes=[models-dir models-pvc-3 storage-config tc-config etcd-config kube-api-access-pqz7t]: timed out waiting for the condition
+			//  Warning  FailedMount  75s (x9 over 3m22s)  kubelet            MountVolume.SetUp failed for volume "storage-config" : secret "storage-config" not found
+			By("Deleting any PVC entries from the storage-config secret")
+			FVTClientInstance.DeleteStorageConfigSecret()
+			// recreate the storage-config secret without the PVCs
+			FVTClientInstance.CreateStorageConfigSecret(StorageConfigDataMinio)
+
+			// after applying configmap, the runtime pod(s) restart, wait for stability
+			By("Waiting for stable deploy state")
+			WaitForStableActiveDeployState()
+
+			isvcObject = NewIsvcForFVT(isvcFiles[isvcWithPvcNotInStorageConfig])
+
+			FVTClientInstance.PrintPods()
+
+			// after mounting the new PVC the runtime pod(s) restart again, but the ISVC
+			// if not scaleToZero, it could have landed on the previous runtime pod will
```

Comment:
These two lines seem a bit jumbled. `scaleToZero` is enabled in the applied configmap. In that case, can the ISVC still land on the previous runtime pod?

Reply:
sorry, I missed to clean this up. another artifact of the 30 plus iterations/commits trying to get the FVTs to run reliably